### PR TITLE
idempotent `allocateQuery` and `notifyLocalViewChanges`

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: packages/firestore/src/local/local_store.ts [hunk 1]
```diff
@@ -812,24 +809,25 @@ export class LocalStore {
                   QueryPurpose.Listen,
                   txn.currentSequenceNumber
                 );
-                return this.queryCache.addQueryData(txn, queryData);
+                return this.queryCache.addQueryData(txn, queryData).next(() => {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/local/local_store.ts]: I think you can just write this as:

```
 return this.queryCache.addQueryData(txn, queryData).next(() => queryData)
```

File: packages/firestore/src/local/local_store.ts [hunk 2]
```diff
@@ -702,41 +702,40 @@ export class LocalStore {
...
+        );
+        this.queryDataByTarget = this.queryDataByTarget.insert(
+          targetId,
+          updatedQueryData
+        );
+      }
+    }
     return this.persistence.runTransaction(
       'notifyLocalViewChanges',
       'readwrite',
```
Comment (schmidt-sebastian) [inline packages/firestore/src/local/local_store.ts]: Here and below: Can you change this to `readwrite-idempotent`? Note that if you merge in the latest changes from mrschmidt/idempotent, the test runner will automatically run the transaction twice (so you can verify that this works).

Comment (wu-hui) [reply packages/firestore/src/local/local_store.ts]: Done.

File: packages/firestore/src/local/local_store.ts (hunk 1 above)
Comment (wu-hui) [reply packages/firestore/src/local/local_store.ts]: Done.

**Pull request merged**